src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyIotCertificate.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.usecases;
+
+import com.aws.greengrass.clientdevices.auth.api.Result;
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.Certificate;
+import com.aws.greengrass.clientdevices.auth.iot.IotAuthClient;
+import com.aws.greengrass.clientdevices.auth.iot.registry.CertificateRegistry;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+public class VerifyIotCertificate implements UseCases.UseCase<Exception, String> {
+    private static final Logger logger = LogManager.getLogger(VerifyIotCertificate.class);
+
+    private final NetworkState networkState;
+    private final IotAuthClient iotAuthClient;
+    private final CertificateRegistry certificateRegistry;
+
+
+    /**
+     * Register core certificate authority with Greengrass cloud.
+     *
+     * @param networkState        Network state infrastructure
+     * @param iotAuthClient       IoT dataplane API client
+     * @param certificateRegistry cert registry
+     */
+    @Inject
+    public VerifyIotCertificate(NetworkState networkState, IotAuthClient iotAuthClient,
+                                CertificateRegistry certificateRegistry) {
+        this.networkState = networkState;
+        this.iotAuthClient = iotAuthClient;
+        this.certificateRegistry = certificateRegistry;
```

Comment:
This is just a comment but ideally we would inject into this a `CertificateRepository` which would just be an interface and have `certificateRegistry` adhere to that interface then we could swap the implementation any time without having to modify everywhere we inject this